### PR TITLE
Include codesnippets into the Javadoc and exclude them from distribut…

### PR DESCRIPTION
**Pull request by jtulach**
…ion JARs if specified as `'snippetsPattern' : '.*SampleSnippets.*'` in suite.py

Pull request to implement what has been discussed in https://github.com/graalvm/mx/issues/57

### CODE REVIEW

File: mx.py [hunk 1]
```diff
@@ -4772,9 +4777,8 @@ def expand(value, context):
             'developer',
             'url',
             'licenses',
-            'licences',
             'defaultLicense',
-            'defaultLicence',
```
Comment (rschatz) [inline mx.py]: Why did you remove these two keys? What effect does removing them have?
(I'm not familiar with that part of the code... @gilles-duboscq, do you know what this does?)

Comment (gilles-duboscq) [inline mx.py]: Initially we were using the UK spelling. In an mx version we switched to US spelling. There is `mx_compat` support for that so that we can still load older suites.
I don't think we should remove these keys.

If there is a problem such as mx silently ignoring the old version of the keys, then we need a different solution were we first read the `mxversion` and then add the keys with the current spelling to the list of supported keys using `mx_compat`. But that would anyway be work for an other pull request.

Comment (jtulach) [inline mx.py]: I am sorry guys. I thought the keys are duplicated. I haven't noticed the difference in "s" vs. "c". I update my diff.

**Comment by jtulach** [comment]
Now the diff contains only changes related to code snippets. Sorry for the noise and thanks for catching the error.

**Comment by rschatz** [comment]
Please also add a version bump, so users will get a "please update mx" message instead of a cryptic "unsupported suite attribute: snippetsPattern".

**Comment by jtulach** [comment]
All comments addressed. Merging.
